visitz/Visitz/Views/TagView.xaml.cs

```diff
@@ -0,0 +1,69 @@
+using Microsoft.Maui.Controls;
+
+namespace Visitz.Views;
+
+public partial class TagView : ContentView
+{
+    public static readonly new BindableProperty BackgroundColorProperty =
+        BindableProperty.Create(nameof(BackgroundColor), typeof(Color), typeof(TagView));
+    public static readonly BindableProperty TextColorProperty =
+        BindableProperty.Create(nameof(TextColor), typeof(Color), typeof(TagView));
+    public static readonly BindableProperty IconNameProperty =
+        BindableProperty.Create(nameof(IconName), typeof(string), typeof(TagView));
+    public static readonly BindableProperty TextProperty =
+        BindableProperty.Create(nameof(Text), typeof(string), typeof(TagView),
+            propertyChanged: (b, o, n) =>
+            {
+                ((TagView)b).updateLayout();
+            });
+    public static readonly BindableProperty BorderColorProperty =
+        BindableProperty.Create(nameof(BorderColor), typeof(Color), typeof(TagView));
+
+    public new Color BackgroundColor
+    {
+        get => (Color)GetValue(BackgroundColorProperty);
+        set => SetValue(BackgroundColorProperty, value);
+    }
+
+    public Color TextColor
+    {
+        get => (Color)GetValue(TextColorProperty);
+        set => SetValue(TextColorProperty, value);
+    }
+
+    public string IconName
+    {
+        get => (string)GetValue(IconNameProperty);
+        set => SetValue(IconNameProperty, value);
+    }
+
+    public string Text
+    {
+        get => (string)GetValue(TextProperty);
+        set => SetValue(TextProperty, value);
+    }
+
+    public Color BorderColor
+    {
+        get => (Color)GetValue(BorderColorProperty);
+        set => SetValue(BorderColorProperty, value);
+    }
+
+    public TagView()
+    {
+        InitializeComponent();
+    }
+
+    protected override void OnBindingContextChanged()
+    {
+        base.OnBindingContextChanged();
+        UpdateUI();
+    }
+
+    private void UpdateUI()
+    {
+        Icon.IsVisible = (IconName?.Length == 0) ? false : true;
+        SpacingView.IsVisible = ((Icon?.IsVisible ?? false) && Text?.Length > 0);
```

Comment:
Is this SpacingView required? Why not use Grid.ColumnSpacing?

If it is, this expression can be simplified:

```suggestion
        SpacingView.IsVisible = Icon.IsVisible && Text.Length > 0);
```

We guarantee these views exist, so no need for nullables (aka Optionals).